### DISCUSSION TITLE
Cold wallet send address for any chia won

### DISCUSSION THREAD

**Discussion by oxygen** [Chia-Network/chia-blockchain#7937] [Ideas]
For farmers, the wallet is accessible on the same machine by internet enabled processes. Even in more complicated farming setups this is still true for one machine. Funds can easily and readily be forwarded by an intruder to his own wallet.

A feature to automatically and instantly forward **won** chia to a cold wallet receive address would add great security (maybe even introduce a clear distinction between farming wallets and cold wallets, see below).

A **secure** way for the user to know if the cold wallet's address has been reconfigured is mandatory, or else intruders will simply use the feature for their own needs. 
To mitigate the above, perhaps deriving the farming key from the cold wallet's key would make this whole process mandatory, automatic, secure, etc. Unfortunately it would require replotting to make use of.

The feature could be summarily documented (how to backup the cold wallet's mnemonic, etc.) in the GUI for users with less security experience.

**Comment by SlugFiller** [comment]
NFT plots already allow you to choose to target whichever wallet you want. Unfortunately, it also means a foreign intruder can override that address, meaning non-OG plots could be tricked into farming directly towards an attacker address, and the only way to tell is to look at the NFT's status (which is far from verbose in the GUI).

Mind you, rewards from the pool, or from solo farming, can already be configured from the GUI via Farm->Manage Farming Rewards. I haven't tested, but it seems to automatically update any active NFTs. I'm not sure if it requires re-plotting for OG plots.

For solo mining with NFT, simply create your own pool from the cold wallet, and set it as the active pool for your farmer.

The upshot is that you can protect the rewards _until_ the attack, and should probably be able to minimize losses after one, by keeping an eye on things, and seeing if the rewards aren't reaching your chosen target wallet, or if your chosen pool isn't confirming points received. In other words, rather than getting your entire wallet sweeped, you only lose the rewards from a small timeframe.

For OG plots, it's best to plot with the cold wallet's address from the get-go. I can't offer direct instructions for plotting with different farmer and reward addresses, however. It's possible, but you'll have to figure that one out yourself.

**Reply by oxygen** [reply]
It would be great to trust the cold wallet address based on a signature or key derivation, or a potential invader could simply reconfigure the .yaml file surreptitiously and never even have to have to touch the disk with malware (won't have to set startup settings, infect exes, etc). 
That is, he will buffer overflow the chia exe, then write to the config using the current process (which is guaranteed  access and expected to actually write to the config) and then dissapear into the night completely undetected by any kind of antivirus.

Otherwise thanks for a good answer, I didn't know about NFT plots as I am still solo with original plots and will not change.

**Reply by SlugFiller** [reply]
Any form of signature or key derivation, unless it's connected directly to the plots (can't be easily recreated), would have to be stored in a way that's just as easy to overwrite as the yaml. If the computer is compromised, especially considering Chia is, by default, installed in the user folder, and not in an Administrator-protected folder, nothing short of a hardware wallet, or making an attack too difficult (e.g. requiring full re-plot) would help.

Mind you that even if the wallet addresses were all visible and easily human-verifiable, being able to overwrite the Chia executable means being able to work with one address, and display a different one. That's no extra protection.
